### PR TITLE
Fix check warnings in photosynthesis (replaces #2821)

### PR DESCRIPTION
**Pull request by infotroph**
This PR initially contains the same commits as #2821, which can no longer be merged because its upstream fork was deleted. Please see there for a lot of good discussion, not all of which was resolved yet when I opened this PR on 2022-08-17.

Goal of any additional commits: Make the minimal changes to get @moki1202's good work merged; it is OK if not every warning in the photosynthesis package is cleaned up.

### CODE REVIEW

File: modules/photosynthesis/R/Licor.QC.R [hunk 1]
```diff
@@ -51,11 +51,11 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
...
          pch = 20, cex = 2, 
          main = paste("UPDATED", dat$fname[1], "\nclick to undo Outliers"))
-    points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
-    legend("bottomright", legend = c("fail", "unchecked", "pass"), 
+      graphics::points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass"), 
            col = status, pch = 18, cex = 1.5, bty = "n")
     
-    flag <- identify(dat$Ci[sel], dat$Photo[sel])
+    flag <-   graphics::identify(dat$Ci[sel], dat$Photo[sel])
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:58]:
```suggestion
    flag <- graphics::identify(dat$Ci[sel], dat$Photo[sel])
```

File: modules/photosynthesis/R/Licor.QC.R [hunk 2]
```diff
@@ -80,13 +80,13 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
          col = status[dat$QC[sel] + 2],
          pch = 20, cex = 2, 
          main = paste("CLICK ON OUTLIERS\n", dat$fname[1]))
-    points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
-    text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
-    legend("bottomright", legend = c("fail", "unchecked", "pass", "other"), 
+      graphics::points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass", "other"),
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:85]:
```suggestion
    graphics::points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
    graphics::text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
    graphics::legend("bottomright", legend = c("fail", "unchecked", "pass", "other"), 
```

File: modules/photosynthesis/R/Licor.QC.R [hunk 3]
```diff
@@ -80,13 +80,13 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
...
-    text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
-    legend("bottomright", legend = c("fail", "unchecked", "pass", "other"), 
+      graphics::points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass", "other"), 
            col = status, pch = 18, cex = 1.5, bty = "n")
     
     ## flag outliers
-    flag <- identify(c(dat$PARi[sel], ulhc[1]), c(dat$Photo[sel], ulhc[2]))
+    flag <-   graphics::identify(c(dat$PARi[sel], ulhc[1]), c(dat$Photo[sel], ulhc[2]))
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:89]:
```suggestion
    flag <- graphics::identify(c(dat$PARi[sel], ulhc[1]), c(dat$Photo[sel], ulhc[2]))
```

File: modules/photosynthesis/R/Licor.QC.R [hunk 4]
```diff
@@ -101,12 +101,12 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
          col = status[dat$QC[sel] + 2], 
          pch = 20, cex = 2, 
          main = paste("UPDATED", dat$fname[1], "\nclick to undo Outliers"))
-    points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
-    legend("bottomright", legend = c("fail", "unchecked", "pass"), 
+      graphics::points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass"),
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:105]:
```suggestion
    graphics::points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
    graphics::legend("bottomright", legend = c("fail", "unchecked", "pass"), 
```

File: modules/photosynthesis/R/Licor.QC.R [hunk 5]
```diff
@@ -101,12 +101,12 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
...
          main = paste("UPDATED", dat$fname[1], "\nclick to undo Outliers"))
-    points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
-    legend("bottomright", legend = c("fail", "unchecked", "pass"), 
+      graphics::points(dat$PARi[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass"), 
            col = status, pch = 18, cex = 1.5, bty = "n")
     
     ## undo outliers
-    flag <- identify(dat$PARi[sel], dat$Photo[sel])
+    flag <-   graphics::identify(dat$PARi[sel], dat$Photo[sel])
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:109]:
```suggestion
    flag <- graphics::identify(dat$PARi[sel], dat$Photo[sel])
```

File: modules/photosynthesis/R/Licor.QC.R [hunk 6]
```diff
@@ -51,11 +51,11 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
          col = status[dat$QC[sel] + 2],
          pch = 20, cex = 2, 
          main = paste("UPDATED", dat$fname[1], "\nclick to undo Outliers"))
-    points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
-    legend("bottomright", legend = c("fail", "unchecked", "pass"), 
+      graphics::points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass"),
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:55]:
```suggestion
    graphics::points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
    graphics::legend("bottomright", legend = c("fail", "unchecked", "pass"), 
```

File: modules/photosynthesis/R/Licor.QC.R [hunk 7]
```diff
@@ -30,13 +30,13 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
...
-    text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
-    legend("bottomright", legend = c("fail", "unchecked", "pass", "other"),
+    graphics::points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass", "other"),
            col = status, pch = 18, cex = 1.5, bty = "n")
     
     ## flag outliers
-    flag <- identify(c(dat$Ci[sel], ulhc[1]), c(dat$Photo[sel], ulhc[2]))
+    flag <-   graphics::identify(c(dat$Ci[sel], ulhc[1]), c(dat$Photo[sel], ulhc[2]))
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:39]:
```suggestion
    flag <- graphics::identify(c(dat$Ci[sel], ulhc[1]), c(dat$Photo[sel], ulhc[2]))
```

File: modules/photosynthesis/R/Licor.QC.R [hunk 8]
```diff
@@ -30,13 +30,13 @@ Licor_QC <- function(dat, curve = c("ACi", "AQ"), tol = 0.05) {
          col = status[dat$QC[sel] + 2], 
          pch = 20, cex = 2, 
          main = paste("CLICK ON OUTLIERS\n", dat$fname[1]))
-    points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
-    text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
-    legend("bottomright", legend = c("fail", "unchecked", "pass", "other"),
+    graphics::points(dat$Ci[-sel], dat$Photo[-sel], col = status[4], pch = 20, cex = 2)
+      graphics::text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
+      graphics::legend("bottomright", legend = c("fail", "unchecked", "pass", "other"),
```
Comment (infotroph) [inline modules/photosynthesis/R/Licor.QC.R:35]:
```suggestion
    graphics::text(ulhc[1], ulhc[2], "FAIL\nALL", col = "red")
    graphics::legend("bottomright", legend = c("fail", "unchecked", "pass", "other"),
```

**Comment by moki1202** [comment]
@infotroph what would be the git commands if I wanted to push changes here?

**Comment by infotroph** [comment]
> what would be the git commands if I wanted to push changes here?

@moki1202 two options: 

a. Push directly to this branch. I *think* you have permission to do this already; if not I can find out how to grant it.
  - add my fork as a remote if you haven't already: `git remote add infotroph https://github.com/infotroph/pecan.git`
  - check out a local branch that tracks mine: `git fetch infotroph && git switch -c moki-patch-16` (I should have given the branch a more meaningful name, but now that the PR is open GitHub won't let me rename it)
  - check out a new branch to work from locally: `git switch -c photosynthesis-check-warnings`
  - commit changes in the `photosynthesis-check-warnings` branch
  - when ready to push, 
      * switch back to the branch that tracks this PR: `git switch moki-patch-16`
      * make sure you have the latest remote changes: `git pull infotroph moki-patch-16`
      * merge in your changes: `git merge photosynthesis-check-warnings`
      * push back: `git push infotroph moki-patch-16`
  - Pro: Simple, adds to the PR immediately without needing to wait for me to do anything.
  - Con: Needs more care to make sure we both keep remote changes pulled into our local branches.

b. Push to your fork and contribute to this branch via PR. Steps that differ from option a are in bold.
  - add my fork as a remote if you haven't already: `git remote add infotroph https://github.com/infotroph/pecan.git`
  - check out a local branch that tracks mine: `git fetch infotroph && git switch -c moki-patch-16` (I should have given the branch a more meaningful name, but now that the PR is open GitHub won't let me rename it)
  - check out a new branch to work from locally: `git switch -c photosynthesis-check-warnings`
  - commit changes in the `photosynthesis-check-warnings` branch
  - **push changes to your fork as you go**: `git push origin photosynthesis-check-warnings` (this assumes you have an `origin` remote that points to github.com/moki1202/pecan.git; if it's named something other than `origin` then use that name)
  - **when you're ready to contribute your changes, open `https://github.com/infotroph/pecan/compare/moki-patch-16...moki1202:pecan:photosynthesis-check-warnings` in your browser**. This will take you directly to the page for opening a pull request from your branch (`moki1202:photosynthesis-check-warnings`) into mine (`infotroph:moki-patch-16`). Click "Create pull request", add a title and description, and click "create pull request" again. (There's a way to get to this page through a series of UI actions too, but they're tedious to write out and the URL works well once you know that the pattern is `github.com/`[target owner]`/`[target repo]`/compare/`[target branch]`...`[source owner]`:`[source repo]`:`[source branch].) 
  - **Optional but useful**: Ping me here to make sure I saw the PR -- I have a history of missing the notifications for things that happen in my fork rather than in the PEcAnProject upstream.
  - Pro: Less care needed to keep branches in sync, alows easy review and discussion as we add things
  - Con: Your changes won't show up in *this* PR, the one we actually care about, until I remember to review and merge the PR you created.

**Comment by Aariq** [comment]
> @infotroph what would be the git commands if I wanted to push changes here?

A lot of this is automated by the `usethis` package.  I think you can do `usethis::pr_fetch(3015)` and it does everything in option a that @infotroph suggested.  You'll just need to make commits and push changes.

**Review by mdietze — APPROVED** [review]

**Comment by infotroph** [comment]
@mdietze Since your last review I added a pile of edits to the plotting code in the vignette. I had to shrink the figures a _lot_ to get its filesize below the magic 5-MB threshold where `R CMD check` stops complaining -- can you please build the vignette locally and confirm it looks acceptable to you?

**Pull request merged**